vscode-sbcl-clpm.md

```diff
@@ -0,0 +1,472 @@
+         (:api-version "0.3")
+
+         (:source "quicklisp"
+                 :url "https://quicklisp.common-lisp-project-index.org/"
+                 :type :ql-clpi)
+
+         (:asd "experiment.asd")
+
+         ;; Dev Dependencies
+         ;;; Swank for that dank repl dev
+         (:system :swank)
+         ~~~
+      - Have clpm download swank and put it in the right place for this
+        project.
+         ~~~lisp
+         cplm bundle install
+         ~~~
+   2. Start up a sbcl
+      - Note: _You can start sbcl and auto-load this project with the
+        asdf integration by running_
+        ```sh
+        rlwrap clpm bundle exec -- sbcl
+        ```
+        _That will read the clpmfile and set up your context correctly.
+        Keep in mind that your context only makes your libraries
+        available, it doesn't actually load them.  You use ASDF to load
+        libraries.  This means that having something like swank as a dev
+        dependency only puts swank in your runtime if you actually load
+        it somewhere. If you were feeling REALLY wild, you could skip
+        straight to step 7 by starting it with this:_
+        ```sh
+        clpm exec --context=experiment -- sbcl \
+        --eval '(setf clpm-client:*asdf-system-not-found-behavior* :install)' \
+        --eval '(setf clpm-client:*context-diff-approval-method* t)' \
+        --eval '(asdf:load-system :swank)' \
+        --eval '(swank:create-server)' \
+        --eval '(asdf:load-system :experiment)
+        ```
+        _That would actually have it automatically load your project,
+        load swank, start a repl and automatically install missing
+        dependencies when it encounters them. I am putting the full
+        instructions here to keep things moderately consistent, though.
+        Feel free to experiment._
+      ```bash
+      rlwrap sbcl
+      ```
+   3. Activate clpm for this project
+      ~~~lisp
+      (clpm-client:activate-context #p"/Users/me/Desktop/experiment/clpmfile" :activate-asdf-integration t)
+      ~~~
+   4. Load swank
+      ```bash
+      (asdf:load-system :swank)
+      ```
+   5. Start up a swank server
+      ~~~lisp
+      (swank:create-server)
+      ~~~
+   6. Load up the experiment project from earlier
+      ~~~lisp
+      (asdf:load-system :experiment)
+      ~~~
+   7. Attach vscode to the repl using the VsCode Command Palette.
+      - `View/Command Palette`
+      - `Alive: Attach to REPL`
+   8. You now have vscode attached to the swank server, which attaches
+      your editor to a running lisp compiler... in this case, a compiler
+      with your project loaded.  You can run commands in the regular CLI
+      repl and the repl in vscode, but where it gets more interesting is
+      when you load files directly from the project and evaluate parts
+      of the code as you work. Check out [this
+      video](https://vimeo.com/496601063) to see a really trite example
+      of what I mean.
```

Comment:
Phew! So these were the steps to have the editor.

Now it would be great to have an example of how the editor works… have a quick explanation of how to load code, how to send code to the REPL, how to use the interactive debugger? Do you mind adding something about that, even quick?

Reply:
I agree with the call for brevity and the previous comments. In general, the gold standard is the loop page.
https://lispcookbook.github.io/cl-cookbook/iteration.html

Short, to the point, and the recipes are structured so that one follows along without distraction, from simple to complex.

* Link rot is a serious concern, long-term. Ideally include text of information needed, and only that, to make the Cookbook page as stand-alone and brief as possible, and  link to a public repo or a resource that is known to be up for a long time. Blogs and private domains for individual developers rot the fastest, from observation.
* Otherwise, great initiative, thanks.

Reply:
I will work on trimming this down and removing the external links.  I will also add a section on using the editor to start a REPL, evaluating some code with an obvious bug, and dropping into the debugger.  I can make a gif if you think it's valuable.  I think moving images of REPL use are powerful, but I wasn't sure if it was reasonable to upload a gif as an asset.